### PR TITLE
Entity framework

### PR DESCRIPTION
**Pull request by cataraul**

### CODE REVIEW

**Review by MariaCrucin — APPROVED** [review]

**Review by AlinCuc — APPROVED** [review]

File: TweetBook/Controllers/V1/PostsController.cs [hunk 1]
```diff
@@ -16,32 +16,35 @@ public PostsController(IPostService postService)
         }
 
         [HttpGet(ApiRoutes.Posts.GetAll)]
-        public IActionResult GetAll()
+        public async Task<IActionResult> GetAll()
         {
-            return Ok(_postService.GetPosts());
+
```
Comment (tatarflavia) [inline TweetBook/Controllers/V1/PostsController.cs:21]: Please remember that between the first parenthesis and the first instruction there is no empty line. Same thing about the end parenthesis of a method and the instruction before it. Please take the time to look through this file and make as many changes as needed.

File: TweetBook/Data/DataContext.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.EntityFrameworkCore;
+using TweetBook.Domain;
+
+namespace TweetBook.Data
+{
+    public class DataContext :DbContext
+    {
+        public DataContext(DbContextOptions<DataContext> options) : base(options)
+        {
+        }
```
Comment (tatarflavia) [inline TweetBook/Data/DataContext.cs:10]: Leave a line between the methods

File: TweetBook/Data/DataContext.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+        public DbSet<Post> Posts { get; set; }
```
Comment (tatarflavia) [inline TweetBook/Data/DataContext.cs:11]: Any new table you add in the db, you put here in the same way another DbSet. A good thing to remember.

File: TweetBook/Domain/Post.cs [hunk 4]
```diff
@@ -1,7 +1,12 @@
-﻿namespace TweetBook.Domain
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace TweetBook.Domain
 {
     public class Post
     {
+
+        [Key]
+
```
Comment (tatarflavia) [inline TweetBook/Domain/Post.cs:9]: The [Key] is called the key attribute. We put attributes between "[ ]" symbols. This attribute here tells us that the Id field from this class is the unique identifier for the Post db table(primary key). One attribute is usually strictly linked to its parameter, so please keep them close and erase the line between.

File: TweetBook/Services/IPostService.cs [hunk 5]
```diff
@@ -4,12 +4,14 @@ namespace TweetBook.Services
 {
     public interface IPostService
     {
-       public List<Post> GetPosts();
+        public Task<List<Post>> GetPostsAsync();
```
Comment (tatarflavia) [inline TweetBook/Services/IPostService.cs:7]: Empty space in front of 'public'

File: TweetBook/Services/IPostService.cs [hunk 6]
```diff
@@ -4,12 +4,14 @@ namespace TweetBook.Services
...
 
-       public Post GetPostById(Guid postId);
+       public Task<Post> GetPostByIdAsync(Guid postId);
 
-       public bool UpdatePost(Post postToUpdate);
+        public Task<bool> CreatePostAsync(Post post);
```
Comment (tatarflavia) [inline TweetBook/Services/IPostService.cs:11]: Empty space in front of 'public'

File: TweetBook/Services/IPostService.cs [hunk 7]
```diff
@@ -4,12 +4,14 @@ namespace TweetBook.Services
...
 
-       public bool DeletePost(Guid postId);
+        public Task<bool> UpdatePostAsync(Post postToUpdate);
```
Comment (tatarflavia) [inline TweetBook/Services/IPostService.cs:13]: Empty space in front of 'public'

**Pull request merged**